test_runner/fixtures/neon_fixtures.py

```diff
@@ -1580,7 +1580,17 @@ def pageserver_start(
             s3_env_vars = self.env.remote_storage.access_env_vars()
             extra_env_vars = (extra_env_vars or {}) | s3_env_vars
 
-        return self.raw_cli(start_args, extra_env_vars=extra_env_vars)
+        try:
+            return self.raw_cli(start_args, extra_env_vars=extra_env_vars)
+        except Exception:
+            # A common reason for startup failure is that the port is already in use. We
+            # coordinate port assignment with PortDistributor, but it's a common mistake
+            # when writing a new test to use a hardcoded port, or assign the port without
+            # using the distributor, causing races where two tests runnign concurrently
+            # sometimes choose the same port. To help debug such cases, get a listing
+            # of all inuse ports and the processes holding them.
+            list_inuse_ports()
+            raise
```

Comment:
Maybe do that for other processes as well? `MockS3Server`, `pg_start`, `NeonPageserver.start`, `VanillaPostgres.start`, `NeonProxy`... Some may be already covered indirectly.

Alternative proposal: `list_inuse_ports` on several `subprocess.run` calls. Or maybe look for a string like "port" in stdout/stderr.